### PR TITLE
Allow `@JsonString` to be used in methods/functions eg in Retrofit services

### LINKED ISSUE #82: Allow `JsonString` to be used in Retrofit
This is for those cases where you want the json string of a response. 

The 'usual' way would be to define your service method return type as `ResponseBody` and call `ResponseBody.string()` at the call site to get the json string but this slight indirection is not necessary
```kotlin
interface Service {

  @GET("the/good/stuff") suspend fun goodStuff(): ResposeBody
}
// myService.goodStuff().string()
```

Being able to just annotate the method with `@JsonString` with a return type of `String` is much better. 

Right now the `@JsonString` annotation targets are `PROPERTY, FIELD`. Adding `FUNCTION` to that should fix this with no other changes.

### PR DESCRIPTION
**Pull request by efemoney**
Fixes #82

### CODE REVIEW

File: moshi-adapters/src/test/kotlin/dev/zacsweers/moshix/adapters/JsonStringTest.kt [hunk 1]
```diff
@@ -67,4 +74,39 @@ class JsonStringTest {
...
+    //language=JSON
+    val json = "{\"a\":2,\"b\":3,\"c\":[1,2,3]}"
+
+    val moshi = Builder()
+      .add(JsonString.Factory())
+      .build()
+
+    val aService = Retrofit.Builder()
+      .baseUrl("http://example.com/")
+      .callFactory(IdempotentCallFactory(json))
```
Comment (ZacSweers) [inline moshi-adapters/src/test/kotlin/dev/zacsweers/moshix/adapters/JsonStringTest.kt:89]: Are you familiar with MockWebServer from OkHttp? I'd prefer to use that over the Callfactory approach here

Comment (efemoney) [reply moshi-adapters/src/test/kotlin/dev/zacsweers/moshix/adapters/JsonStringTest.kt:89]: Yeah I can switch it to MockWebServer. Will update the PR in a bit.

Comment (efemoney) [reply moshi-adapters/src/test/kotlin/dev/zacsweers/moshix/adapters/JsonStringTest.kt]: Done!

**Review by ZacSweers — APPROVED** [review]

**Comment by ZacSweers** [comment]
Thanks! I should have another release out shortly

**Pull request merged**